### PR TITLE
feat: update programmatic oruga

### PR DESCRIPTION
**Pull request by mlmoravek**
I updated the programmatic interface for oruga by adding type definitions and usability enhancements. 

**Contains Breaking Changes**

## Proposed Changes

- Added programmatic interface type definitions
- Removed wrapping layer for programmatic interface
- Changed `useProgrammatic` to `useOruga`

### CODE REVIEW

**Comment by jtommy** [comment]
I agree with you but it's a breaking change, so we can release it only on next 0.x version

**Comment by mlmoravek** [comment]
> I agree with you but it's a breaking change, so we can release it only on next 0.x version

Yeah, no need to merge it right away. We can wait until we have more for a major release update.

File: packages/oruga-next/src/components/loading/index.ts [hunk 1]
```diff
@@ -3,40 +3,47 @@ import { createVNode, render } from "vue";
...
 import {
     registerComponent,
     registerComponentProgrammatic,
-} from "../../utils/plugins";
-import InstanceRegistry from "../../utils/InstanceRegistry";
+} from "@/utils/plugins";
+import InstanceRegistry from "@/utils/InstanceRegistry";
+
+declare module "@/types" {
+    interface OrugaPrgrammatic {
```
Comment (messenjer) [inline packages/oruga-next/src/components/loading/index.ts:15]: There is this typo `OrugaPrgrammatic` in a few places

Comment (mlmoravek) [reply packages/oruga-next/src/components/loading/index.ts]: @messenjer I appreciate the review. Typo is fixed :)

**Pull request merged**